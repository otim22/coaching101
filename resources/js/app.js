require('./bootstrap');

require('./components/initialFormator');
