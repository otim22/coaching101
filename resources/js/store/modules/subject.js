import subjectService from '../../services/subjectService.js'

export const namespaced = true

// State
export const state = {
    subjects: [],
    subjectItem: {
        introduction: {
            subject_title: null,
            subject_subtitle: null,
            subject_description: null,
            default_subject: '-- Subject --',
            subject_options: [
                'Mathematics', 'Science', 'English', 'Chemistry','Biology', 'Swahili', 'French', 'Agriculture',
                'Food & nutrition', 'Social Studies', 'CRE', 'IRE','Geography','Entreprenuership', 'Commerce',
                'Accounts', 'Economics', 'Divinity','History'
            ],
            default_class: '-- Class --',
            class_options: ['Senior one', 'Senior two', 'Senior three', 'Senior four', 'Senior five', 'Senior six'],
            default_level: '-- Level --',
            level_options: ['Term one', 'Term two', 'Term three'],
            done: false,
        },
        outlines: [
            {
                content_title: null,
                main_content_files: null,
                content_description: null,
                extra_resource_files: null,
                done: false
            }
        ],
        students_learn: [
            {
                students_learn: null,
                done: false
            }
        ],
        class_requirement: [
            {
                class_requirement: null,
                done: false
            }
        ],
        target_student: [
            {
                target_student: null,
                done: false
            }
        ],
        subject_message: {
            welcome_message: null,
            congratulations_message: null,
            done: false
        }
    }
}

// Mutations
export const mutations = {
    addSubject(state, subject) {
        state.subjects = [...state.subjects, subject]
    },
    getSubject(state, subject) {
        state.subject = subject
    },
    removeSubject(state, subject) {
        state.subjects.splice(state.subjects.indexOf(subject), 1)
    }
}

// Actions
export const actions = {
    createSubject({ commit, dispatch }, subject) {
        return subjectService.postSubject(subject)
                .then((res) => {
                    commit('addSubject', res.data)
                    const notification = {
                        type: 'success',
                        message: 'Your subject has been created!'
                    }
                    dispatch('notification/add', notification, { root: true })
                })
                .catch(error => {
                    const notification = {
                        type: 'error',
                        message: 'There was a problem creating your subject: ' + error.message
                    }
                    dispatch('notification/add', notification, { root: true })
                    throw error
                })
    }
}

// Getters
export const getters = {
    subjects: state => state.subjects,
    getSubjectById: state => id => {
        return state.subjects.find(subject => subject.id === id)
    }
}
