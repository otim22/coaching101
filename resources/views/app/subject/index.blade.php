@extends('layouts.app')

@section('content')

<section class="section-two mt-5">
    <div class="container">
        <div class="row mt-5">
            <div class="col-lg-8 offset-2">
                <h2>Subjects</h2>
                @foreach($subjects as $subject)
                <a href="{{ route('subjects.show', $subject) }}">  {{ $subject->title }}</a><br />
                @endforeach
            </div>
        </div>
    </div>
</section>

@endsection
