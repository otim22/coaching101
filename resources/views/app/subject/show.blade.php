@extends('layouts.app')

@section('content')

<section class="section-two mt-5">
    <div class="container">
        <div class="row mt-5">
            <div class="col-lg-8 offset-2">
                <h3>{{ $subject->title }}</h3>
                <p>Subtitle {{ $subject->subtitle }} </p>
                <p>Description {{ $subject->description }} </p>
                <p>Image </p>
                <img src="{{ asset($subject->getFirstMediaUrl()) }}" class="w-50">

                <ul>
                    @foreach($subject->audience['student_learn'] as $student_learn)
                    <li>Learn {{ $student_learn }}</li>
                    @endforeach
                </ul>

                <ul>
                    @foreach($subject->audience['class_requirement'] as $class_requirement)
                    <li>Reqirements {{ $class_requirement }}</li>
                    @endforeach
                </ul>

                <ul>
                    @foreach($subject->audience['target_student'] as $target_student)
                    <li>Target {{ $target_student }}</li>
                    @endforeach
                </ul>

                <p>Welcome {{ $subject->message['welcome_message'] }} </p>
                <p>Kudos {{ $subject->message['congragulation_message'] }} </p>
                <a href="{{ route('topics', $subject) }}">Create Topic</a><br />
                <a href="{{ route('all-subjects') }}">All Subjects</a>
            </div>
        </div>
    </div>
</section>

@endsection
