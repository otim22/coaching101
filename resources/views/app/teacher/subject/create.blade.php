@extends('layouts.app')

@section('content')
<section>
    <class-creation />
</section>
@endsection
