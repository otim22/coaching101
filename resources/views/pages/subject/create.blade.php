@extends('layouts.app')

@section('content')

<section class="section-two mt-5">
    <div class="container">
        <div class="row mt-5">
            <div class="col-lg-8 offset-2">
                <form action="{{ route('subjects') }}" method="POST" enctype="multipart/form-data">
                    @csrf

                    <div class="fast-transition mb-3">
                        <div class="row m-2 pb-2">
                            <div class="col-sm-12 col-md-12 col-lg-12">
                                <h3>Subject introduction</h3> <hr />
                            </div>

                            <div class="col-sm-12 col-md-12 col-lg-12">
                                <div class="form-group">
                                    <label for="title">Subject title</label>
                                    <div class="input-group">
                                        <input type="text"
                                                    class="form-control @error('title') is-invalid @enderror"
                                                    id="title"
                                                    placeholder="Your subject title"
                                                    aria-label="Your subject title"
                                                    aria-describedby="title"
                                                    name="title"
                                                    value="{{ old('title') }}"
                                                    required>
                                    </div>
                                    @error('title')
                                        <div class="alert alert-danger p-2 mt-2">{{ $message }}</div>
                                    @enderror
                                </div>

                                <div class="form-group">
                                    <label for="subtitle">Sub-title</label>
                                    <div class="input-group">
                                        <input type="text"
                                                    class="form-control @error('subtitle') is-invalid @enderror"
                                                    id="subtitle"
                                                    placeholder="Write your sub title"
                                                    aria-label="Write your sub title"
                                                    aria-describedby="subtitle"
                                                    name="subtitle"
                                                    value="{{ old('subtitle') }}">
                                        @error('subtitle')
                                            <div class="alert alert-danger p-2 mt-2">{{ $message }}</div>
                                        @enderror
                                    </div>
                                </div>

                                <div class="form-group">
                                    <label for="description">Description</label>
                                    <textarea class="form-control @error('description') is-invalid @enderror" id="description" placeholder="Description of the subject" name="description" value="{{ old('description') }}" rows="3" required></textarea>
                                    @error('description')
                                        <div class="alert alert-danger p-2 mt-2">{{ $message }}</div>
                                    @enderror
                                </div>

                                <div class="form-group">
                                    <label for="title">Category</label>
                                    <div class="input-group">
                                        <input type="text"
                                                    class="form-control @error('category') is-invalid @enderror"
                                                    id="category"
                                                    placeholder="Eaxmple: Modern physics"
                                                    aria-label="category"
                                                    aria-describedby="category"
                                                    name="category"
                                                    value="{{ old('category') }}" required>
                                    </div>
                                    @error('category')
                                        <div class="alert alert-danger p-2 mt-2">{{ $message }}</div>
                                    @enderror
                                </div>

                                <div class="form-group">
                                    <label for="cover_image">Cover Image</label>
                                    <input type="file" name="cover_image" class="form-control-file @error('cover_image') is-invalid @enderror" id="cover_image" required>
                                    @error('cover_image')
                                        <div class="alert alert-danger p-2 mt-2">{{ $message }}</div>
                                    @enderror
                                </div>
                            </div>
                        </div>
                    </div>

                    <div class="col-sm-12 col-md-12 col-lg-12 d-flex justify-content-between mt-5">
                        <div>
                            <h5>Step 1 of 3</h5>
                        </div>
                        <div>
                            <button type="submit" class="btn btn-primary btn-block btn-md pl-5 pr-5 ml-3 mr-3">Save</button>
                        </div>
                    </div>
                </form>
            </div>
        </div>
    </div>
</section>

@endsection
