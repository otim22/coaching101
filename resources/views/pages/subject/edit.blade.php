@extends('layouts.app')

@section('content')

<section class="section-bread bg-gray-4">
    <div class="container">
        <nav aria-label="breadcrumb bg-gray">
            <ol class="breadcrumb">
                <li class="breadcrumb-item"><a style="text-decoration: none" href="{{ url('/') }}">Home</a></li>
                <li class="breadcrumb-item"><a style="text-decoration: none" href="{{ route('teacher.subjects') }}">Subjects</a></li>
                <li class="breadcrumb-item active" aria-current="page">{{ $subject->short_title }}</li>
            </ol>
        </nav>
    </div>
</section>

<section class="section-two">
    <div class="container">
        <div class="row mt-5 mb-4">
            <div class="col-lg-8 offset-2">
                <form action="{{ route('subjects.update', $subject) }}" method="POST" enctype="multipart/form-data">
                    @csrf
                    @method('PATCH')

                    <div class="fast-transition mb-3">
                        <div class="row m-2 pb-2">
                            <div class="col-sm-12 col-md-12 col-lg-12">
                                <h3>Subject introduction</h3> <hr />
                            </div>
                            <div class="col-sm-12 col-md-12 col-lg-12">
                                <div class="form-group mb-4 mt-3">
                                    <label for="title" class="bold">Subject title</label>
                                    <div class="input-group">
                                        <input type="text"
                                                    class="form-control @error('title') is-invalid @enderror"
                                                    id="title"
                                                    placeholder="Your subject title"
                                                    aria-label="Your subject title"
                                                    aria-describedby="title"
                                                    name="title"
                                                    value="{{ old('title', $subject->title) }}"
                                                    required>
                                    </div>
                                    @error('title')
                                        <div class="alert alert-danger p-2 mt-2">{{ $message }}</div>
                                    @enderror
                                </div>

                                <div class="form-group mb-4">
                                    <label for="subtitle" class="bold">Sub-title</label>
                                    <div class="input-group">
                                        <input type="text"
                                                    class="form-control @error('subtitle') is-invalid @enderror"
                                                    id="subtitle"
                                                    placeholder="Write your sub title"
                                                    aria-label="Write your sub title"
                                                    aria-describedby="subtitle"
                                                    name="subtitle"
                                                    value="{{ old('subtitle', $subject->subtitle) }}">
                                        @error('subtitle')
                                            <div class="alert alert-danger p-2 mt-2">{{ $message }}</div>
                                        @enderror
                                    </div>
                                </div>

                                <div class="form-group mb-4">
                                    <label for="description" class="bold">Description</label>
                                    <textarea class="form-control @error('description') is-invalid @enderror" id="description" placeholder="Description of the subject" name="description" rows="3" required>{{ old('description', $subject->description) }}</textarea>
                                    @error('description')
                                        <div class="alert alert-danger p-2 mt-2">{{ $message }}</div>
                                    @enderror
                                </div>

                                <div class="form-group mb-4">
                                    <label for="title" class="bold">Category</label>
                                    <div class="input-group">
                                        <input type="text"
                                                    class="form-control @error('category') is-invalid @enderror"
                                                    id="category"
                                                    placeholder="Eaxmple: Modern physics"
                                                    aria-label="category"
                                                    aria-describedby="category"
                                                    name="category"
                                                    value="{{ old('category', $subject->category) }}" required>
                                    </div>
                                    @error('category')
                                        <div class="alert alert-danger p-2 mt-2">{{ $message }}</div>
                                    @enderror
                                </div>

                                <div class="form-group">
                                    <p class="bold">Current Image</p>
                                    <img src="{{ asset($subject->getFirstMediaUrl()) }}" class="mb-4 rounded-corners w-100">
                                    <p><small style="color: red; font-weight: bold;">Choosing Another Image Replaces Current Image</small></p>

                                    <label for="cover_image" class="bold">Cover Image</label>
                                    <input type="file" name="cover_image" class="form-control-file @error('cover_image') is-invalid @enderror" id="cover_image">
                                    @error('cover_image')
                                        <div class="alert alert-danger p-2 mt-2">{{ $message }}</div>
                                    @enderror
                                </div>
                            </div>
                        </div>
                    </div>

                    <div class="col-sm-12 col-md-12 col-lg-12 d-flex justify-content-between mt-5">
                        <div>
                            <a href="{{ route('subjects.show', $subject) }}" class="btn btn-secondary btn-block pl-5 pr-5">
                                <svg width="1.3em" height="1.3em" viewBox="0 0 20 20" class="bi bi-box-arrow-in-left" fill="currentColor" xmlns="http://www.w3.org/2000/svg">
                                    <path fill-rule="evenodd" d="M10 3.5a.5.5 0 0 0-.5-.5h-8a.5.5 0 0 0-.5.5v9a.5.5 0 0 0 .5.5h8a.5.5 0 0 0 .5-.5v-2a.5.5 0 0 1 1 0v2A1.5 1.5 0 0 1 9.5 14h-8A1.5 1.5 0 0 1 0 12.5v-9A1.5 1.5 0 0 1 1.5 2h8A1.5 1.5 0 0 1 11 3.5v2a.5.5 0 0 1-1 0v-2z"/>
                                    <path fill-rule="evenodd" d="M4.146 8.354a.5.5 0 0 1 0-.708l3-3a.5.5 0 1 1 .708.708L5.707 7.5H14.5a.5.5 0 0 1 0 1H5.707l2.147 2.146a.5.5 0 0 1-.708.708l-3-3z"/>
                                </svg>
                                Back
                            </a>
                        </div>
                        <div>
                            <button type="submit" class="btn btn-primary btn-block pl-5 pr-5 ml-3 mr-3">Update</button>
                        </div>
                    </div>
                </form>
            </div>
        </div>
    </div>
</section>
@endsection
