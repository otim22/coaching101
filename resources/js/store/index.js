import Vue from 'vue'
import Vuex from 'vuex'
import * as subject from './modules/subject.js'
import * as notification from './modules/notification.js'

Vue.use(Vuex);

export default new Vuex.Store({
    modules: {
        subject,
        notification
    }
})
