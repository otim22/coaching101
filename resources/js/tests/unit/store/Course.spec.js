import { mount } from '@vue/test-utils'
import axios from 'axios';
import * as subject from '../../../store/modules/subject.js'

describe('subject/getters', () => {
    it('should mount without crashing', () => {
        const wrapper = mount(subject)
        expect(wrapper).toMatchSnapshot()
    })

    it('it returns subjects', () => {
        const state = {
            subjects: [{
                introduction: {
                    subject_title: 'Subject title',
                    subject_subtitle: 'Subject subtitle',
                    subject_description: 'Subject description',
                },
                outlines: [
                    {
                        content_title: 'Content title',
                        main_content_files: [],
                        done: false
                    }
                ],
                students_learn: [
                    {
                        students_learn: 'Student learn',
                        done: false
                    }
                ],
                class_requirement: [
                    {
                        class_requirement: 'Class requirement',
                        done: false
                    }
                ],
                target_student: [
                    {
                        target_student: 'Target student',
                        done: false
                    }
                ],
                subject_message: [
                    {
                        welcome_message: 'Welcome message',
                        congratulations_message: 'Congratulations message',
                        done: false
                    }
                ]
            }]
        }

        expect(subject.getters.subjects(state)).toBe(state.subjects)
    })
})

describe('subject/mutations', () => {
    test('addSubject adds a new subject', () => {
        const state = {
            subjects: []
        }
        const subjectItem = [{
            introduction: {
                subject_title: 'Subject title',
                class_options: [
                    "Senior one",
                    "Senior two",
                    "Senior three",
                    "Senior four",
                    "Senior five",
                    "Senior six",
               ],
               subject_description: "subject_description",
               subject_subtitle: "subject_subtitle",
               subject_title: "subject_title",
               default_class: "-- Class --",
               default_level: "-- Level --",
               default_subject: "-- Subject --",
               done: false,
               level_options: [
                   "Term one",
                   "Term two",
                   "Term three",
               ],
               subject_options: [
                    "Mathematics",
                    "Science",
                    "English",
                    "Chemistry",
                    "Biology",
                    "Swahili",
                    "French",
                    "Agriculture",
                    "Food & nutrition",
                    "Social Studies",
                    "CRE",
                    "IRE",
                    "Geography",
                    "Entreprenuership",
                    "Commerce",
                    "Accounts",
                    "Economics",
                    "Divinity",
                    "History",
               ],
                subject_subtitle: "subject_subtitle",
                subject_description: "subject_description",
            },
            outlines: [
                {
                    content_title: "content_title",
                    content_description: "content_description",
                    main_content_files: [],
                    extra_resource_files: [],
                    done: false
                }
            ],
            students_learn: [
                {
                    students_learn: "students_learn",
                    done: false
                }
            ],
            class_requirement: [
                {
                    class_requirement: "class_requirement",
                    done: false
                }
            ],
            target_student: [
                {
                    target_student: "target_student",
                    done: false
                }
            ],
            subject_message: [
                {
                    welcome_message: "welcome_message",
                    congratulations_message: "congratulations_message",
                    done: false
                }
            ]
        }]
        subject.mutations.addSubject(state, subjectItem[0])
        expect([...state.subjects]).toMatchObject(subjectItem)
    })
})

/* To be continued*/
describe('subject/actions', () => {
    test('createSubject commits addSubject mutation', () => {
        jest.mock('axios');
        const commit = jest.fn()
        const subjectItem = [{
            introduction: {
                subject_title: 'Subject title',
                class_options: [
                    "Senior one",
                    "Senior two",
                    "Senior three",
                    "Senior four",
                    "Senior five",
                    "Senior six",
               ],
               subject_description: "subject_description",
               subject_subtitle: "subject_subtitle",
               subject_title: "subject_title",
               default_class: "-- Class --",
               default_level: "-- Level --",
               default_subject: "-- Subject --",
               done: false,
               level_options: [
                   "Term one",
                   "Term two",
                   "Term three",
               ],
               subject_options: [
                    "Mathematics",
                    "Science",
                    "English",
                    "Chemistry",
                    "Biology",
                    "Swahili",
                    "French",
                    "Agriculture",
                    "Food & nutrition",
                    "Social Studies",
                    "CRE",
                    "IRE",
                    "Geography",
                    "Entreprenuership",
                    "Commerce",
                    "Accounts",
                    "Economics",
                    "Divinity",
                    "History",
               ],
                subject_subtitle: "subject_subtitle",
                subject_description: "subject_description",
            },
            outlines: [
                {
                    content_title: "content_title",
                    content_description: "content_description",
                    main_content_files: [],
                    extra_resource_files: [],
                    done: false
                }
            ],
            students_learn: [
                {
                    students_learn: "students_learn",
                    done: false
                }
            ],
            class_requirement: [
                {
                    class_requirement: "class_requirement",
                    done: false
                }
            ],
            target_student: [
                {
                    target_student: "target_student",
                    done: false
                }
            ],
            subject_message: [
                {
                    welcome_message: "welcome_message",
                    congratulations_message: "congratulations_message",
                    done: false
                }
            ]
        }]

        axios.post('/subjects', subjectItem)
            .then(data => {
                subject.actions.createSubject({ commit }, subjectItem)
                expect(commit).toHaveBeenCalledWith('addSubject', 1)
            })
    })
})
