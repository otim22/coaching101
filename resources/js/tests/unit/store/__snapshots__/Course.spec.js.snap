// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subject.js should mount without crashing 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;

exports[`subject/getters should mount without crashing 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;
